business/checkers/common/check_confirmer.go

```diff
@@ -0,0 +1,18 @@
+package common
+
+import (
+	"fmt"
+
+	"github.com/kiali/kiali/models"
+)
+
+// ConfirmIstioCheckMessage can be used by the validation tests to confirm the Istio Check received is what was expected
+func ConfirmIstioCheckMessage(expectedIstioCheckKey string, actualIstioCheck *models.IstioCheck) error {
```

Comment:
I generally don't like to put test code next to production code. So I'd try to find a place in a `_test.go` file. For me it is a bit confusing. If you are alright with the suggestion, it should definitely be under `business` or `business/checkers`. What about creating the `business/checkers/checker_test.go` (which is where the base interfaces are defined) and placing there both the logic and its test together?

Reply:
I tried every way I could think possible - the first thing I tried was exactly what you suggest (I created business/checkers/checker_test.go and it didn't work), and then I tried business/checkers/common/xyz_test.go - again, doesn't work). Nothing worked without getting `import cycle not allowed in test` errors. If you can figure out how to put test code in _test files that work across package boundaries without getting a circular dependency error in golang, please tell me how :) I googled and everybody says it's impossible.

Reply:
I've seen a few places having a `testing.go` file or some `testing`, `util/testing`, `testhelpers`... package to keep common test helpers. If they are never imported into the non-testing code I'm not sure if they get included in the final binary or not.

Reply:
Right - they are still "outside" of the testing framework (just like I have it) but because they aren't used outside of _test code, it shouldn't get compiled in the final binary. So the way this PR has it right now, I believe this test function won't be in the final binary because it isn't used outside of test code (but that is only going by what people say in the Internets - I really have no idea).

Note that the name is irrelevant (people use "test" or "testing" just to make it human-understandable what it is for - but Go makes no distinction (e.g. see "httptest").

Reply:
@nrfox thanks for passing by. I was thinking of asking you to review this part since you've got some experience of go outside of this repo.

Okay, I get the `import cycle not allowed in test` problem. Perhaps aligned with @nrfox we could name the file differently to make it explicit that it contains test-only code. Otherwise, I see the common package and I think that each file in there is one checker (common - meaning that applies to different object while the folder destinationrules, for instance, only contains checkers exclusively thought for destinationrules. But each file listed is one checker. So it'd be good to keep the consistency). Then perhaps rename the file to `testing` and `testing_test` or `testhelpers` and `testhelpers_test` or even creating a generic folder for the utils.

Reply:
Wait. we already have a test package with helpers for the test: https://github.com/kiali/kiali/tree/master/tests
In the data folder there are a bunch of helpers for creating fixtures for the test. We might one to create a utils folder with the helpers you created. wdyt?

Reply:
I did notice that tests folder.. It actually has things in the data/ subdirectory - I wasn't sure what it was for. If that is where we put stuff in there, I should be able to move that file from common to here. I'll try it and see.

Reply:
Ahh.. yes, the test_assert.go - I tried playing with that one (that's in test/data/validations) - I was getting those weird dependency errors in there, too. But that was early on in my coding. I'll revisit this. Not everything in this test/data directory involves creating test data - this test_assert.go is one example. But I won't refactor this stuff - I'll just try to put something in here - perhaps in the test/data/validations directory.

Reply:
OK, I can already see problems with moving it under test/data/validations. I'm going to do this:

> We might one to create a utils folder with the helpers you created. 

I'll put a `testutil` under tests/ directory.

Reply:
OK, see latest commit - moved that stuff out into a new `testutils` package located directly under the `tests` directory.

Reply:
Sound good to me. The `testutils` package sounds the place to be.